### PR TITLE
feat(build): maven refactor for maven central

### LINKED ISSUE #79: publish sdk to sonatype central
We should look to move off of github packages to support anonymous reads. I started this pr a while back but it required more work than expected to get setup. 

https://github.com/opentdf/java-sdk/pull/25

See https://virtru.atlassian.net/wiki/spaces/ENG/pages/3543105546/Maven+Central+Sonatype

### LINKED ISSUE #173: Add BSD-3-Clear license to repository
please see other public repositories in opentdf for example. 

add LICENSE to the main directory of the repo.

### PR DESCRIPTION
**Pull request by pflynn-virtru**
- Deploy to Maven Central (only `sdk`) see https://central.sonatype.org/publish/publish-portal-maven/
- Removes support for OSSRH and Nexus. see https://central.sonatype.org/register/legacy/
- Maven `develop` profile (default) builds `sdk` and `cmdline`
- Maven `stage` profile deploys `sdk` to GitHub Packages  (only main branch)
- Maven `release` profile deploys `sdk` to Maven Central (only on release, [no snapshots](https://central.sonatype.org/faq/snapshot-releases/))
- `sdk/target/generated-sources` replaces `sdk/src/main/protogen`
- adds Maven configuration around license, etc. see https://central.sonatype.org/publish/requirements/#supply-javadoc-and-sources
  - sources and checksums sdk jar
  - LICENSE

Resolves #173 
Resolves #79

### CODE REVIEW

**Comment by ttschampel** [comment]
Can you do a local test publish of the snapshot to snapshot repo?

I would think you'd need some of the additions to the pom outlined here : https://central.sonatype.org/publish/publish-maven/#distribution-management-and-authentication 

e.g. 
```
<plugin>
                <groupId>org.sonatype.plugins</groupId>
                <artifactId>nexus-staging-maven-plugin</artifactId>
                <extensions>true</extensions>
                <configuration>
                    <stagingProgressTimeoutMinutes>15</stagingProgressTimeoutMinutes>
                    <serverId>repository.apache.org</serverId>
                    <nexusUrl>https://repository.apache.org/</nexusUrl>
                </configuration>
            </plugin>
```

**Review by dmihalcik-virtru — APPROVED** [review]

**Comment by pflynn-virtru** [comment]
> Can you do a local test publish of the snapshot to snapshot repo?
No, we removed OSSRH and Nexus support

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -43,8 +45,41 @@ jobs:
...
+          </server></servers>
+          </settings>
+          EOF
+          mvn --batch-mode deploy -DskipTests -P stage
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          BUF_INPUT_HTTPS_USERNAME: opentdf-bot
+          BUF_INPUT_HTTPS_PASSWORD: ${{ secrets.PERSONAL_ACCESS_TOKEN_OPENTDF }}
+          MAVEN_GPG_PASSPHRASE: ${{ secrets.GPG_KEY_PASSPHRASE }}
+      - name: Publish to Maven Central
```
Comment (ttschampel) [inline .github/workflows/release.yaml:68]: I think we probably should publish snapshots too?  when on main branch?

Comment (ttschampel) [reply .github/workflows/release.yaml:68]: I guess with removal of sonatype - no snapshots are supported :(

**Review by strantalis — APPROVED** [review]

**Pull request merged**